### ISSUE TITLE
Is it appropriate to memoize classnames?

### ISSUE THREAD

**Issue by pconerly** [JedWatson/classnames#128]
React renders a lot, and my team is wondering if it'd be appropriate to memoize `classnames`.  I'm trying to think of the use cases, and it seems like we can always expect consistent output for a given input.

Is memoizing classnames worth it? I'm wondering if memoize's cost of comparing args costs just as much as classname's manipulation.

**Comment by newyork-anthonyng** [comment]
@pconerly Great question. I set up a quick [example repo](https://github.com/newyork-anthonyng/memoize-classnames-test).

It's a basic example with only one component, so take the results with a grain of salt. 
* render time **with** memoization to be around `2.5ms`
* render time **without** memoization to be below `1ms`.

My guess for this performance is the overhead of creating a hash key for the arguments going into `classNames`. In my example, I used `JSON.stringify`.
